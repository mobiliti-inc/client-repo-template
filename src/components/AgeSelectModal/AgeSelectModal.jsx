import React from 'react';

import { iconInsurance } from '../../assets/icons';
import Modal from '../Modal/Modal';
import Dropdown from '../Dropdown/Dropdown';
import styles from './AgeSelectModal.scss';
import Button from '../Button/Button';


let ages = [
	{
		value: '21',
		current: false
	},
	{
		value: '22',
		current: false
	},
	{
		value: '23',
		current: false
	},
	{
		value: '24',
		current: false
	},
	{
		value: '25+',
		current: true
	}
];
class AgeSelectDropdown extends React.PureComponent {
	static propTypes = {
		callback: PropTypes.func.isRequired,
		scrolling: PropTypes.bool.isRequired,
		selectUserAge: PropTypes.func.isRequired
	}

	state = {
		modalVisible: true,
		selectedAge: '25+'
	}

	onContinue = async () => {
		const { selectedAge } = this.state;
		this.setState({
			modalVisible: false
		});
		await this.props.selectUserAge(selectedAge);
		this.props.callback();
	}

	ageSelected = (value) => {
		ages = ages.map(age => ({
			value: age.value,
			current: age.value === value
		}));

		this.setState({
			selectedAge: value
		});
	}

	render() {
		const { modalVisible } = this.state;
		return (
			<React.Fragment>
				<Modal
					visible={modalVisible}
					plain
					scrolling={this.props.scrolling}
					freezeOverlay
				>
					<div styleName="content-container">
						<div styleName="icon-circle-wrapper">
							<img src={iconInsurance} alt="lock" />
						</div>
						<div styleName="heading" >
							Insurance is included in each car&#39;s monthly subscription price.
						</div>

						<p styleName="description">
							Enter your age so we can give you an accurate price.
						</p>
						<Dropdown modal data={ages} onChange={this.ageSelected} className="modal-option" scrolling={false} />

						<div styleName="continue-button-container">
							<Button
								onClick={this.onContinue}
							> Continue
							</Button>
						</div>
					</div>
				</Modal>
			</React.Fragment>
		);
	}
}

export default CSSModules(AgeSelectDropdown, styles);
