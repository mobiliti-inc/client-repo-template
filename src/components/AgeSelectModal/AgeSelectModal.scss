@import "~variables";


.content-container {
    height: 462px;
    display: grid;
    flex-direction: column;
    @media (max-width: $mobile-width) {
        height: 100%;
        margin-top: 100px;
    }
}

.heading {
    color: $darker-blue;
    font-family: $futura;
    font-size: 28px;
    font-weight: bold;
    line-height: 31px;
    margin-bottom: 0;
    margin-top: 10px;
    width: 100%;
}

.description {
    color: $darker-blue;
    font-family: $hanken-sans;
    font-size: 16px;
    line-height: 19px;
    margin: 0;
    padding-top: 5px;
    width: 100%;
    font-weight: normal;
}

.icon-circle-wrapper {
    @include icon-circle-wrapper-mix;
}

.continue-button-container {
    width: 50%;
    cursor: pointer;
    @media (max-width: $mobile-width) {
        width: 100%;
    }
}