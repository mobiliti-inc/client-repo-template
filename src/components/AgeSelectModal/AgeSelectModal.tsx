import React from 'react';
import CSSModules from 'react-css-modules';

// import { iconInsurance } from '../../assets';
import { Modal, Dropdown, Button } from '../';

import styles from './AgeSelectModal.scss';

let ages = [
	{
		value: '21',
		current: false
	},
	{
		value: '22',
		current: false
	},
	{
		value: '23',
		current: false
	},
	{
		value: '24',
		current: false
	},
	{
		value: '25+',
		current: true
	}
];

type AgeSelectDropdownProps = {
	callback: (...args: any[]) => any,
	scrolling: boolean,
	selectUserAge: (...args: any[]) => any
};

type AgeSelectDropdownState = {
	modalVisible: boolean,
	selectedAge: string
};

class AgeSelectDropdown extends React.PureComponent<AgeSelectDropdownProps, AgeSelectDropdownState> {
	state = {
		modalVisible: true,
		selectedAge: '25+'
	};

	onContinue = async () => {
		const { selectedAge } = this.state;
		this.setState({
			modalVisible: false
		});
		await this.props.selectUserAge(selectedAge);
		this.props.callback();
	}

	ageSelected = value => {
		ages = ages.map(age => ({
			value: age.value,
			current: age.value === value
		}));
		this.setState({
			selectedAge: value
		});
	}

	render() {
		const { modalVisible } = this.state;
		return (
			<React.Fragment>
				<Modal visible={modalVisible} plain scrolling={this.props.scrolling} freezeOverlay>
					<div styleName="content-container">
						<div styleName="icon-circle-wrapper">
							<img src={iconInsurance} alt="lock" />
						</div>
						<div styleName="heading">Insurance is included in each car&#39;s monthly subscription price.</div>

						<p styleName="description">Enter your age so we can give you an accurate price.</p>
						<Dropdown modal data={ages} onChange={this.ageSelected} className="modal-option" scrolling={false} />

						<div styleName="continue-button-container">
							<Button onClick={this.onContinue}> Continue</Button>
						</div>
					</div>
				</Modal>
			</React.Fragment>
		);
	}
}

export default CSSModules(AgeSelectDropdown, styles);
