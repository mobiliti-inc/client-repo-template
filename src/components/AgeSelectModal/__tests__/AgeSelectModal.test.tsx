import AgeSelectModal from '../AgeSelectModal';

describe('AgeSelectModal', () => {
	it('renders', () => {
		const subject = shallow(<AgeSelectModal
			scrolling
			callback={() => { }}
			selectUserAge={() => { }}
		/>);

		expect(subject.text()).toContain('<Modal />');
		matchSnapshot(subject);
	});
});
