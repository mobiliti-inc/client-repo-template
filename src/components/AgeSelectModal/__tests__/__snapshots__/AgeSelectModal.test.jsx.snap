// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgeSelectModal renders 1`] = `
<React.Fragment>
  <Modal
    bodyClassName=""
    buttonClassName={null}
    buttonText=""
    footerClassName=""
    freezeOverlay={true}
    headerClassName=""
    icon={<div />}
    modalParentClass=""
    noIcon={false}
    noPointer={false}
    onButtonClick={[Function]}
    onIconClose={[Function]}
    overlay=""
    plain={true}
    plainHeader={false}
    scrolling={true}
    showButtonLoader={false}
    showCloseIcon={false}
    subTitle=""
    title=""
    visible={true}
  >
    <div
      className="content-container"
    >
      <div
        className="icon-circle-wrapper"
      >
        <img
          alt="lock"
          src="test-file-stub"
        />
      </div>
      <div
        className="heading"
      >
        Insurance is included in each car's monthly subscription price.
      </div>
      <p
        className="description"
      >
        Enter your age so we can give you an accurate price.
      </p>
      <Dropdown
        className="modal-option"
        customOptionsList={null}
        customSelected={null}
        data={
          Array [
            Object {
              "current": false,
              "value": "21",
            },
            Object {
              "current": false,
              "value": "22",
            },
            Object {
              "current": false,
              "value": "23",
            },
            Object {
              "current": false,
              "value": "24",
            },
            Object {
              "current": true,
              "value": "25+",
            },
          ]
        }
        dropDownMenuTitleClass=""
        dropdownParentClass=""
        dropdownSelectedClass=""
        header={null}
        modal={true}
        onChange={[Function]}
        scrolling={false}
      />
      <div
        className="continue-button-container"
      >
        <Button
          bordered={false}
          buttonType="standard"
          className=""
          customStyles=""
          disabled={false}
          icon={null}
          inverted={false}
          onClick={[Function]}
          ripple={false}
          showLoader={false}
        >
           Continue
        </Button>
      </div>
    </div>
  </Modal>
</React.Fragment>
`;
